src/components/Header/components/PeopleSearch/index.js

```diff
@@ -215,20 +215,22 @@ export default class GordonPeopleSearch extends Component {
                 ),
               ].map((e, key) => <span key={key}>{e}</span>)
             : this.getHighlightedText(
-                suggestion.Nickname &&
-                  suggestion.Nickname !== suggestion.FirstName &&
-                  suggestion.Nickname !== suggestion.UserName.split(/ |\./)[0]
-                  ? suggestion.FirstName + ' (' + suggestion.Nickname + ') ' + suggestion.LastName
-                  : suggestion.MaidenName &&
-                    suggestion.MaidenName !== suggestion.LastName &&
-                    suggestion.MaidenName !== suggestion.UserName.split(/ |\./)[1]
-                  ? suggestion.FirstName +
-                    ' ' +
-                    suggestion.LastName +
-                    ' (' +
-                    suggestion.MaidenName +
-                    ')'
-                  : suggestion.FirstName + ' ' + suggestion.LastName,
+                suggestion.FirstName +
+                // If having nickname that is unique, display that nickname
+                (suggestion.Nickname &&
+                suggestion.Nickname !== suggestion.FirstName &&
+                suggestion.Nickname !== suggestion.UserName.split(/ |\./)[0]
+                ? ' (' + suggestion.Nickname + ') '
+                : ' ') +
+                suggestion.LastName +
+                // If having maiden name that is unique, display that maiden name
+                (suggestion.MaidenName &&
+                suggestion.MaidenName !== suggestion.LastName &&
+                suggestion.MaidenName !== suggestion.UserName.split(/ |\./)[1]
+                ? ' (' +
+                suggestion.MaidenName +
+                ')'
+                : ''),
                 this.state.highlightQuery,
```

Comment:
This is a large number of expressions in the return of the component; I wonder if all of this should be in a function somewhere else? You could call `getHighlightedText(getSuggestedName(suggestion))` or something like that. But it's just an idea to make it neater, I don't know that it's entirely necessary. What do you think?

Reply:
(We discussed this in person; not worth doing for now.)